lib/deps.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+const kModuleError = Symbol('moduleError');
+
+function makeErrorModule(error) {
+  return new Proxy(
+    {},
+    {
+      get: (_, key) => {
+        if (key === kModuleError) {
+          return error ? error : new Error('Module Not Loaded');
+        }
+        throw error;
+      },
+      set: () => {
+        throw error;
+      }
+    }
+  );
+}
+
+let BSON = require('bson');
+try {
+  BSON = require('bson-ext');
+} catch (_) {} // eslint-disable-line
+
+let Kerberos = makeErrorModule(new Error('`kerberos` module not found. Install it or disable it.'));
```

Comment:
Now that we're here, maybe we can improve these error messages. How about:
"Optional module `keberos` not found. Please install it to enable kerberos authentication"

This would be different for each module, and maybe we could even provide a link to a docs page on installing dependencies?